generator/lib/control.ml

```diff
@@ -419,56 +451,45 @@ let api =
           methods = [
             {
               Method.name = "probe";
-              description = "[probe uri]: looks for existing SRs on the storage device";
+              description = String.concat " " [
+                "[probe configuration]: can be used iteratively to narrow down configurations";
+                "to use with SR.create, or to find existing SRs on the backing storage";
+              ];
               inputs = [
-                uri;
+                configuration;
               ];
               outputs = [
                { Arg.name = "result";
-                 ty = 
-                   Type.Struct (
-                     ("srs", Type.(Array sr_stat), "SRs found on this storage device"), [
-                      "uris", Type.(Array (Basic String)), "Other possible URIs which may contain SRs"
-                   ]);
+                 ty = Type.(Array (Name "probe_result"));
                  description = "Contents of the storage device";
                }
               ];
             };
             {
               Method.name = "create";
-              description = "[create uuid uri name description configuration]: creates a fresh SR";
+              description = "[create uuid configuration name description]: creates a fresh SR";
               inputs = [
                 uuid;
-                uri;
+                configuration;
                 { Arg.name = "name";
                   ty = Type.(Basic String);
                   description = "Human-readable name for the SR";
                 }; {
                   Arg.name = "description";
                   ty = Type.(Basic String);
                   description = "Human-readable description for the SR";
-                }; {
-                  Arg.name = "configuration";
-                  ty = Type.(Dict(String, Basic String));
-                  description = String.concat " " [
-                    "Plugin-specific configuration which describes where and";
-                    "how to create the storage repository. This may include";
-                    "the physical block device name, a remote NFS server and";
-                    "path or an RBD storage pool.";
-                  ];
                 };
               ];
-              outputs = [uri]
+              outputs = [configuration]
             };
             {
               Method.name = "attach";
               description = String.concat " "[
-                "[attach uuid uri]: attaches the SR to the local host. Once an SR";
+                "[attach configuration]: attaches the SR to the local host. Once an SR";
```

Comment:
So now UUID is part of the `configuration`?

Reply:
For SR.attach yes, the reason was that you should get this information from SR.probe, so you do an SR.probe, get a `device-config` map, and then just call attach with that.
SR.create is different because the user would be supplying the `device-config`, but the UUID will be generated by XAPI.